File: app/src/main/java/isaacborges/com/projetoconcrete/retrofit/services/RepositoriosService.java
package isaacborges.com.projetoconcrete.retrofit.services;

import isaacborges.com.projetoconcrete.dto.PullRequestSync;
import isaacborges.com.projetoconcrete.dto.RepositoriosSync;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface RepositoriosService {

    @GET("search/repositories")
    Call<RepositoriosSync> buscaRepositorios(
            @Query("q") String linguagem,
            @Query("sort") String ordem,
            @Query("page") int numeroDaPagina);

    @GET("repos/{autor}/{repositorio}/pulls")
    Call<PullRequestSync> buscaPullRequestsDoRepositorio(@Path("autor") String nomeAutor, @Path("repositorio") String nomeRepositorio);

}

File: app/src/main/java/br/com/eriberto/desafioandroidconcrete/model/retrofit/services/RepositoriosService.kt
package br.com.eriberto.desafioandroidconcrete.model.retrofit.services

import br.com.eriberto.desafioandroidconcrete.model.pojo.RepositorioDAO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface RepositoriosService {

    @GET("search/repositories?q=language:Java&sort=stars")//"search/repositories?q=language:Java&sort=stars&page=1"
    fun getRepositorios(
        @Query("page") page: Int
    ): Call<RepositorioDAO>
}